### DISCUSSION TITLE
Set log level manually

### DISCUSSION THREAD

**Discussion by bfaccruent** [open-telemetry/opentelemetry-dotnet#4048] [Q&A]
Hi

I am using OpenTelemetry in my ASP.NET API app and I wonder where does it read the log level and can I override it ?

**Comment by cijothomas** [comment]
LogLevel is from ILogger. 
https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.ilogger.log?view=dotnet-plat-ext-7.0#microsoft-extensions-logging-ilogger-log-1(microsoft-extensions-logging-loglevel-microsoft-extensions-logging-eventid-0-system-exception-system-func((-0-system-exception-system-string)))

Typically controlled with choice of methods like LogInformation vs LogWarning vs LogError etc.

**Comment by behdad088** [comment]
Hi @bfaccruent  In an ASP.NET application using OpenTelemetry, the log level is typically configured through the logging framework used by the application, such as Serilog or NLog. By default, OpenTelemetry will use the logging framework's default log level, which is typically Information.

To override the log level used by OpenTelemetry, you can configure the logging framework to use a different log level. For example, if you are using Serilog, you can set the log level by adding the following configuration to your appsettings.json file:

```
{
  "Serilog": {
    "MinimumLevel": {
      "Default": "Debug"
    }
  }
}

``` 

This will set the log level for all loggers to Debug. You can also set the log level for specific loggers by using their logger name in the configuration. For example:

``` 
{
  "Serilog": {
    "MinimumLevel": {
      "Default": "Information",
      "Microsoft": "Warning",
      "System": "Error"
    }
  }
}

``` 
This will set the log level to Information for all loggers, except for the Microsoft logger which will use Warning, and the System logger which will use Error.

Once you have configured the logging framework, OpenTelemetry will use the log level set by the logging framework.

**Reply by cijothomas** [reply]
Serilog is not involved here. (There is an experimental Serilog adapter, but thats not yet shipped.)
For log filtering, use the standard filtering mechanism from Ms.Extensions.Logging itself.
